examples/disaggregated/slurm/benchmark/submit_dwdp.py

```
@@ -208,6 +208,72 @@ def submit_dwdp_job(config, log_dir, dry_run):
+    for _role in ("CTX", "GEN"):
+        for _sid in sorted(allocations.get(_role, {}).keys()):
+            for _host, _gpus in allocations[_role][_sid]["nodes"].items():
+                _per_node.setdefault(_host, []).append((_role, _sid, _gpus))
+    print("  Per-node placement:")
+    for _host in sorted(_per_node.keys()):
+        _ctxs = [(s, g) for r, s, g in _per_node[_host] if r == "CTX"]
+        _gens = [(s, g) for r, s, g in _per_node[_host] if r == "GEN"]
+        _parts = []
+        if _ctxs:
+            _parts.append(
+                f"CTX {[s for s, _ in _ctxs]} (GPU {sorted(sum([g for _, g in _ctxs], []))})"
```

Ruff (0.15.14)

[warning] Avoid quadratic list summation

Replace with `functools.reduce`

(RUF017)